meerk40t/core/cutplan.py

```diff
@@ -639,6 +643,88 @@ def clear(self):
         self.plan.clear()
         self.commands.clear()
 
+    def optimize_rasters(self, operation_list, op_type):
+        def generate_clusters(operation):
+            clusters = list()
+            cluster_bounds = list()
+            for node in operation.children:
+                if node.type == "reference":
+                    node = node.node
+                try:
+                    bb = node.paint_bounds
+                except AttributeError:
+                    continue
+                clusters.append([node])
```

Comment:
In my original code I think I had the node and its bbox as a tuple in a single array. There is no reason why they can't be held separately like this, but it does mean that the cluster merges need to be done to the two arrays in sync. and that is a tiny bit more likely to go astray than holding them together as a tuple.

With a bit more insight into modern coding techniques, I think that if I was recoding the 0.7 version now I would use a data-object to hold a cluster rather than a tuple.